### PR TITLE
Sanitize event

### PR DESCRIPTION
**Pull request by ludeeus**

### CODE REVIEW

File: supervisor/bootstrap.py [hunk 1]
```diff
@@ -319,6 +321,25 @@ def filter_data(event, hint):
...
+        if event.get("request"):
+            if event["request"].get("url"):
+                event["request"]["url"] = sanitize_url(event["request"]["url"])
+
+            for header in event["request"].get("headers", []):
+                if header[0] == "Referer":
+                    header[1] = sanitize_url(header[1])
+
+                if header[0] == "X-Hassio-Key":
+                    header[1] = DUMMY_VALUE
```
Comment (pvizeli) [inline supervisor/bootstrap.py:338]: We use that only once, let's just write "xxxxx" without a global

File: supervisor/bootstrap.py [hunk 2]
```diff
@@ -319,6 +321,25 @@ def filter_data(event, hint):
...
+
+        if event.get("request"):
+            if event["request"].get("url"):
+                event["request"]["url"] = sanitize_url(event["request"]["url"])
+
+            for header in event["request"].get("headers", []):
+                if header[0] == "Referer":
+                    header[1] = sanitize_url(header[1])
+
+                if header[0] == "X-Hassio-Key":
```
Comment (pvizeli) [inline supervisor/bootstrap.py:337]: All kind of headers which we use are in `const.py`. There are more around

File: supervisor/bootstrap.py [hunk 3]
```diff
@@ -319,6 +321,25 @@ def filter_data(event, hint):
...
+                    header[1] = DUMMY_VALUE
+
+                if header[0] in ["Host", "X-Forwarded-Host"]:
+                    event["request"]["headers"].remove(header)
```
Comment (pvizeli) [inline supervisor/bootstrap.py:341]: Let not remove it, we need it for fix ingress, but it into header dummy replacement above

File: supervisor/utils/__init__.py [hunk 4]
```diff
@@ -7,8 +7,9 @@
 import socket
 from typing import Optional
 
+from ..const import RE_STRING, RE_URL
+
 _LOGGER: logging.Logger = logging.getLogger(__name__)
-RE_STRING = re.compile(r"\x1b(\[.*?[@-~]|\].*?(\x07|\x1b\\))")
```
Comment (pvizeli) [inline supervisor/utils/__init__.py:11]: Do we need this global?

File: supervisor/const.py [hunk 5]
```diff
@@ -16,6 +17,9 @@
 
 SUPERVISOR_DATA = Path("/data")
 
+RE_STRING = re.compile(r"\x1b(\[.*?[@-~]|\].*?(\x07|\x1b\\))")
+RE_URL: re.Pattern = re.compile(r"(\w+:\/\/)(.*\.\w+)(.*)")
```
Comment (pvizeli) [inline supervisor/const.py:21]: See comment above

File: supervisor/bootstrap.py (hunk 3 above)
Comment (ludeeus) [reply supervisor/bootstrap.py:341]: These 2 only contains the hostname like "mydomain.com", just displaying "example.com" seems unneeded?

File: supervisor/bootstrap.py (hunk 2 above)
Comment (ludeeus) [reply supervisor/bootstrap.py:337]: Good idea

File: supervisor/utils/__init__.py (hunk 4 above)
Comment (ludeeus) [reply supervisor/utils/__init__.py:11]: Probably not

File: supervisor/const.py (hunk 5 above)
Comment (ludeeus) [reply supervisor/const.py:21]: See comment above

File: supervisor/bootstrap.py (hunk 3 above)
Comment (pvizeli) [reply supervisor/bootstrap.py:341]: Is more to have the full header around and not hunting ghosts

File: supervisor/const.py [hunk 6]
```diff
@@ -63,6 +63,9 @@
 CONTENT_TYPE_URL = "application/x-www-form-urlencoded"
 COOKIE_INGRESS = "ingress_session"
 
+HEADER_FORWARDED_HOST = "X-Forwarded-Host"
+HEADER_HOST = "Host"
+HEADER_REFERER = "Referer"
```
Comment (pvizeli) [inline supervisor/const.py:68]: Let's use for the official headers the hdrs module as we do on core and other parts i.e. (https://github.com/home-assistant/supervisor/blob/dev/supervisor/api/ingress.py#L8)

File: supervisor/utils/filter.py [hunk 7]
```diff
@@ -0,0 +1,87 @@
+"""Filter utils."""
```
Comment (pvizeli) [inline supervisor/utils/filter.py:1]: let's move this into `misc` because `utils` are unaware of coresys.

**Pull request merged**